testdata/src/a/a.go

```diff
@@ -155,3 +155,19 @@ func f13[T int | int8 | int16 | int32 | int64 | uint | uint8 | uint16 | uint32 |
 
 	return b
 }
+
+/* ----------------- issue 21 ----------------- */
+
+func f16(ctx context.Context, k string) func() {
+	return func() { // want "Function `f16\\$1` should pass the context parameter"
+		f16(context.Background(), k)
+	}
+}
+
+func f17(ctx context.Context, k string) func() func() {
+	return func() func() { // want "Function `f17\\$1->f17\\$1\\$1` should pass the context parameter"
+		return func() {
+			f16(context.Background(), k)
+		}
+	}
+}
```

Comment:
Maybe we should add a few cases covering HTTP handlers? Like

```go
func f18(w http.ResponseWriter, r *http.Request) func(w http.ResponseWriter, r *http.Request) {
	return func(w http.ResponseWriter, r *http.Request) {
		f8(context.Background(), w, r) // want "Non-inherited new context, use function like `context.WithXXX` or `r.Context` instead"
	}
}
```

Reply:
This case was already working.